backend/src/test/java/net/wodie/backend/controller/BtoControllerTest.java

```diff
@@ -136,7 +143,58 @@ void getBtoItemStatusById() {
         assertNotNull(actual);
         String expected = initItem1().getStatus();
         Assertions.assertEquals(expected, actual);
+    }
+
+    @Disabled ("Seems to be a problem with embedded MongoDB")
+    @Test
+    void updateBtoVoteTest() {
+        //GIVEN
+        btoRepository.insert(initItem1());
+
+        //WHEN
+        BtoItem actual = testClient.put()
+                .uri("/api/btoItem/vote/"+initItem1().getId())
+                .headers(http -> http.setBearerAuth(jwtToken))
+                .bodyValue(btoVote1())
+                .exchange()
+                .expectStatus().is2xxSuccessful()
+                .expectBody(BtoItem.class)
+                .returnResult()
+                .getResponseBody();
+
+        //THEN
+        assertNotNull(actual);
+        assertTrue(actual.getActionNotMembers().contains("U15"));
+        assertFalse(actual.getActionNotMembers().contains("U15"));
+    }
```

Comment:
TL;DR: I am not sure, is this some error you do not want to have in your merge? In BtoService there is one if: 
```
if (btoTemp.getStatus().equals("VOTE")) {
```
The condition is false for this test, so a
```
throw new ResponseStatusException(HttpStatus.BAD_REQUEST,"Wrong status");
```
is thrown - instead of the here expected 200 status.

My history and how I found it:
> No, this is no problem with mongo. 
> `/*
>     @PutMapping("/vote/{id}")
>     public BtoItem updateBtoVote(@PathVariable String id, @RequestBody BtoVote btoVote) {
>         return btoService.updateBtoVote(id, btoVote);
>     }
>  */
> 
>     @PutMapping("/vote/{id}")
>     public BtoItem stefanPut(@PathVariable String id, @RequestBody BtoVote btoVote) {
>         return btoService.updateBtoVote(id, btoVote);
>     }`
> gives 400.
> 
> ```
> @PutMapping("/vote/{id}")
>     public BtoItem stefanPut(@PathVariable String id, @RequestBody BtoVote btoVote) {
>         return btoService.easyPut(id, btoVote);
>     }
> 
> ```
> gives an assertion error - as expected. Therefore there must be a problem with ` return btoService.updateBtoVote(id, btoVote);`.
> 
> The error message (see right before the EXPECTED/ACTUAL comparison in the test log)
> ```
> {"timestamp":"2022-06-04T20:09:37.430+00:00","status":400,"error":"Bad Request","path":"/api/btoItem/vote/1"}
> ```
> fits to the thrown exception in BtoService line 61:
> ```
> throw new ResponseStatusException(HttpStatus.BAD_REQUEST,"Wrong status");
> ```

Reply:
Das hört sich interessant an, dass werde ich mir morgen in Ruhe noch einmal ansehen

Reply:
Hallo Stefan,
deine Analyse des Tests war korrekt - die Ursache ärgerlich. Ich habe den Test getraced und festgestellt, dass ich den falschen Status in den Test übergebe und daher der Test fehlschlägt.
Der Test läuft jetzt und der easyPutTest wurde dafür entfernt.
Danke nochmals für das gründliche schauen - Wolfgang